### ISSUE TITLE
_FillValue used for some network parameters does not match with _FillValue specified in NetCDF header

### ISSUE THREAD

**Issue by rhutten** [Deltares/HYDROLIB-core#337]
**Describe the bug**
*NOTE (Oct 12, 2022), see updated diagnosis below*

Incorrect _FillValues are specified in the NetCDF header for the parameters: 

- mesh2d_face_nodes -->-2147483648:
![image](https://user-images.githubusercontent.com/55204099/190151542-2f690191-56fc-4484-9063-3aab9d0a55a5.png)

- mesh2d_edge_nodes --> -2147483648
![image](https://user-images.githubusercontent.com/55204099/190151606-ea947727-f4cd-49d9-b339-fd1e46120da7.png)

- link1d2d_contact_type --> -2147483648
![image](https://user-images.githubusercontent.com/55204099/190151375-a852e202-abe9-4283-8804-5dae12b1c7b5.png)

These _FillValues should be -214783648 (netCDF4.default_fillvals['i4']. Else the 1D2D/2D model will not run with a DIMRset. 

(Discovered by running a D-HyDAMO model)

**To Reproduce**

**Expected behavior**
Use integer(non-NaN) _FillValue instead, for example the netcdf default value (available in netCDF4.default_fillvals['i4'] for integers.
Find an alternative for current code:
        mesh2d_fn = ncfile.createVariable(
            "mesh2d_face_nodes",
            "i4",
            (nc_mesh2d.face_dimension, nc_mesh2d.max_face_nodes_dimension),
            fill_value=np.iinfo(np.int32).min,
        )
        mesh2d_en = ncfile.createVariable(
            "mesh2d_edge_nodes",
            "i4",
            (nc_mesh2d.edge_dimension, "Two"),
            fill_value=np.iinfo(np.int32).min,
        )
        link1d2d_contact_type = ncfile.createVariable(
            "link1d2d_contact_type",
            "i4",
            "nLink1D2D_edge",
            fill_value=np.iinfo(np.int32).min,
        )
**Diagnosis**
Fill values have been partially fixed in the meantime, under #290, edge_nodes, face_nodes and also contact_type. However, the factory method for filling the mesh2d connectivity arrays in memory still have a hardcoded fill value, see this piece of code:
```python
    def _process(self, mesh2d_input) -> None:
# [..]
        self.mesh2d_face_nodes = np.full(
            (len(self.mesh2d_face_x), max(npf)), np.iinfo(np.int32).min
        )
```
Coincidentally this "array fill value" is the same value now as the "file variable fill value", but if the user configures a different fill value for the writing (using Tim's new features), then the face_nodes array will be written as wrong values into the file.
This needs fixing.

**Comment by MRVermeulenDeltares** [comment]
@rhutten & @veenstrajelmer Would you have any suggestions how we could test this?

**Comment by MRVermeulenDeltares** [comment]
@rhutten & @tim-vd-aardweg, might this have already been resolved in: #363 ?

**Comment by veenstrajelmer** [comment]
@MRVermeulenDeltares that might indeed be the case.

**Comment by MRVermeulenDeltares** [comment]
Discussed with @rhutten and @tim-vd-aardweg, we will close the issue, it appears to be fixed in #363.
If the problem reoccurs in the future we can create a new issue.
